services/checks/checks.go

```diff
@@ -0,0 +1,206 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+// Package checks provides security checks functionality.
+package checks
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"net"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	api "github.com/percona-platform/saas/gen/checked"
+	"github.com/percona-platform/saas/pkg/check"
+	"github.com/percona/pmm/utils/tlsconfig"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+const (
+	defaultHost     = "check.percona.com"
+	defaultInterval = 24 * time.Hour
+
+	// Environment variables that affect checks service; only for testing.
+	envHost      = "PERCONA_TEST_CHECKS_HOST"
```

Comment:
let's add these new environment variables to our env var parser https://github.com/percona/pmm-managed/blob/45edb31bfcdcd539a6953e93cdd0cd8b0b2c9385/utils/envvars/parser.go#L51

Reply:
I think we should print a BIG warning and skip validation for all `PERCONA_TEST_` variables.

Reply:
These variables are equal to telemetry service test variables (ex. "PERCONA_TEST_TELEMETRY_HOST"), their only purpose to hack behavior for test purposes as checks service hasn't any explicit configuration.

Reply:
Unless we print a BIG warning, someone will rely on them :) (someone will still rely on them, but we did all we could)

Reply:
Big, scary warning added:
https://github.com/percona/pmm-managed/pull/367/files#diff-030e0ff0b16cbb9db111d4f44f1ee2f5R69-R75